### PR TITLE
Issue #SB-15375 feat: User add and read api and test case

### PR DESCRIPTION
**Pull request by souravdey091**

### CODE REVIEW

File: server/controllers/user.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import * as _ from 'lodash';
+import { containerAPI } from "OpenRAP/dist/api";
+
+export default class User {
+    private userSDK;
+    constructor() {
+        this.userSDK = containerAPI.getUserSdkInstance();
+    }
+
+    async addUser(req, res) {
```
Comment (HarishGangula) [inline server/controllers/user.ts]: The methodname should be create

File: server/controllers/user.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            logger.info(`ReqId = "${req.headers['X-msgid']}": request: ${_.get(req, 'body.request')} found from desktop app update api`);
+            return res.send(Response.success('api.desktop.user.create', createResp.id, req));
+        } catch (err) {
+            logger.error(`ReqId = "${req.headers['X-msgid']}": Received error while adding user,  where err = ${err}`);
+            res.status(500);
+            return res.send(Response.error("api.desktop.user.create", 500));
+        }
+    }
+
+    async getUser(req, res) {
```
Comment (HarishGangula) [inline server/controllers/user.ts]: This method name should be read

File: server/controllers/user.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        try {
+            const userData = await this.userSDK.read();
+            logger.info(`ReqId = "${req.headers['X-msgid']}": result: ${userData} found from desktop app update api`);
+            return res.send(Response.success('api.desktop.user.read', userData, req));
```
Comment (harishgilimi) [inline server/controllers/user.ts]: Should  handle 404 cases separately

File: server/routes.ts [hunk 4]
```diff
@@ -563,9 +564,18 @@ export class Router {
...
 
-    let desktopApp = new DesktopApp();
+    let desktopAppUpdate = new DesktopAppUpdate();
     app.get("/api/desktop/v1/update", (req, res) => {
-      desktopApp.getDesktopAppUpate(req, res);
+      desktopAppUpdate.getDesktopAppUpate(req, res);
+    });
+
+    let user = new User();
+    app.post("/api/desktop/user/v1/create", (req, res) => {
```
Comment (HarishGangula) [inline server/routes.ts]: This can be user.addUser.bind(user)

File: server/routes.ts [hunk 5]
```diff
@@ -563,9 +564,18 @@ export class Router {
...
+      user.addUser(req, res);
+    });
+
+    app.get("/api/desktop/user/v1/read", (req, res) => {
```
Comment (HarishGangula) [inline server/routes.ts]: user.getUser.bind(user)

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: server/controllers/user.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+export default class User {
+    private userSDK;
+    constructor() {
+        this.userSDK = containerAPI.getUserSdkInstance();
+    }
+
+    async create(req, res) {
+        try {
+            const createResp = await this.userSDK.create(_.get(req, 'body.request'));
```
Comment (HarishGangula) [inline server/controllers/user.ts:20]: validation required for the request body

File: server/controllers/user.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        }
+    }
+
+    async read(req, res) {
+        try {
+            const userData = await this.userSDK.read();
+            logger.info(`ReqId = "${req.headers['X-msgid']}": result: ${userData} found from desktop app update api`);
+            return res.send(Response.success('api.desktop.user.read', userData, req));
+        } catch (err) {
+            logger.error(`ReqId = "${req.headers['X-msgid']}": Received error while getting user,  where err = ${err}`);
```
Comment (HarishGangula) [inline server/controllers/user.ts:38]: res.status() and res.send status should be same

**Review by HarishGangula — CHANGES_REQUESTED** [review]

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**